### DISCUSSION TITLE
Inaccurate Vulnerability Detection / Inventory

### DISCUSSION THREAD

**Discussion by tkhi** [wazuh/wazuh#20434] [Q&A]
### Environment
- **Wazuh version**: v4.6.0
- **OS version**: Microsoft Windows 10 Pro 10.0.19045.3693

### Description
We have encountered an issue with the Wazuh vulnerability scanner where it incorrectly reports a vulnerability related to an outdated version of WebView2. The Wazuh inventory data shows both version 109 and version 119 of WebView2 present on a specific PC. However, the vulnerability alert pertains to version 109, which, upon manual verification, is not installed on the PC - only version 119 is present.

#### Expected Behavior
The Wazuh vulnerability scanner should accurately reflect the installed versions of software on each system, in this case, only showing WebView2 version 119 for the affected PC.

#### Current Behavior
The vulnerability scanner is indicating that the PC is vulnerable due to an outdated version of WebView2 (version 109). However, this version is not installed on the PC, and only version 119 is present.

### Troubleshooting Steps Already Taken
- Manual verification of the installed WebView2 version on the affected PC.

I seek assistance in troubleshooting why the Wazuh inventory scanner is incorrectly identifying the presence of an older, vulnerable version of WebView2, and how to rectify this issue to ensure accurate vulnerability reporting.
Any guidance or insights into this matter would be greatly appreciated.

Cheers,
Tobiasz Keller

**Comment by pereyra-m** [comment]
Hello @tkhi !

When a package is deleted from an agent, the **syscollector** module informs the vulnerability detector and all the related vulnerabilities to that package should be removed.
Were both packages installed in the host at the same time or it was always just one and an upgrade was performed?

Please, upload the complete agent packages inventory and vulnerabilities inventory so we can know if:
- The package is still being reported by **syscollector**
- The package isn't reported by **syscollector** but vulnerability detector still holds its vulnerabilities.

Regards.

**Comment by tkhi** [comment]
Hello @pereyra-m,

Thank you for your prompt response.

I'm fairly new to Wazuh, so I appreciate your guidance on this matter. Regarding the WebView2 versions on the affected PC, to my knowledge, there was never a simultaneous installation of both versions 109 and 119. However I have no way to verify that.

Attached, please find the complete agent packages inventory and vulnerabilities inventory. From what I can see, both versions are still being reported by the syscollector, which seems unusual since only version 119 is present on the system when checking appwiz.cpl.

[packages.csv](https://github.com/wazuh/wazuh/files/13477574/packages.csv)
Last scan: Nov 27, 2023 @ 16:34:29.000
Version	Architecture	Vendor	Name	agent_id
119.0.2151.72	i686	Microsoft Corporation	Microsoft Edge WebView2-Laufzeit	59
109.0.1518.78	 	Microsoft Corporation	Microsoft Edge WebView2-Laufzeit	59

[vulnerabilities.csv](https://github.com/wazuh/wazuh/files/13477717/vulnerabilities.csv)
Last full scan: Nov 27, 2023 @ 07:32:52.000
Last partial scan: Nov 27, 2023 @ 17:12:01.000

I look forward to your insights on why this might be happening and how to resolve this issue.

Cheers,
Tobiasz

**Comment by pereyra-m** [comment]
Hello again @tkhi !
Thanks for the information provided.

As long as **syscollector** reports the package twice, we'll see the vulnerabilities for both packages. If you look closely, you'll find that they have different versions and architectures.
Now we need to confirm if this situation is an error.

Could you share a screenshot of your `Uninstall or change a program` section of the control panel of the affected host? I'd like to confirm if Windows reports only once application or not.

I've locally tried to reproduce this issue with the installers published here
https://developer.microsoft.com/en-us/microsoft-edge/webview2/?ch=1#download

However, I was unable to install two versions with different architectures, I always receive the message that the application is already installed.

Finally, could you share the output of this debugging tool? The **sysinfo** component collects the agent information behind **syscollector**.
You should run `sysinfo_test_tool.exe --packages`

[testSysinfo.zip](https://github.com/wazuh/wazuh/files/13488663/testSysinfo.zip)


Regards.

**Comment by tkhi** [comment]
Thank you, @pereyra-m.

The problem is now resolved:
![grafik](https://github.com/wazuh/wazuh/assets/152165819/457fa7dc-e5c8-4606-8fb6-f4a8386ff467)

I took this screenshot about 2 hours after the vulnerability was shown as active, so I can't say for sure if the second version was still installed at the time the scanner did its job:
![grafik](https://github.com/wazuh/wazuh/assets/152165819/8f87b5e8-533f-466a-bc85-ae4bb495f4f2)

Let's wait and see if the problem emerges on other devices.

**Reply by pereyra-m** [reply]
Great news @tkhi !
